File: tools/content-verify/__fixtures__/tracks/demo/problems/lc-9004-design/solution.go
package main

type MinStack struct {
	items    []int
	minimums []int
}

func Constructor() MinStack {
	return MinStack{}
}

func (this *MinStack) Push(val int) {
	this.items = append(this.items, val)
	if len(this.minimums) > 0 && this.minimums[len(this.minimums)-1] < val {
		val = this.minimums[len(this.minimums)-1]
	}
	this.minimums = append(this.minimums, val)
}

func (this *MinStack) Pop() {
	this.items = this.items[:len(this.items)-1]
	this.minimums = this.minimums[:len(this.minimums)-1]
}

func (this *MinStack) Top() int {
	return this.items[len(this.items)-1]
}

func (this *MinStack) GetMin() int {
	return this.minimums[len(this.minimums)-1]
}

File: tools/content-verify/__fixtures__/tracks/demo/problems/lc-9004-design/Solution.java
import java.util.*;

class MinStack {
    private final Deque<Integer> items = new ArrayDeque<>();
    private final Deque<Integer> minimums = new ArrayDeque<>();

    public MinStack() {
    }

    public void push(int val) {
        items.push(val);
        minimums.push(minimums.isEmpty() ? val : Math.min(val, minimums.peek()));
    }

    public void pop() {
        items.pop();
        minimums.pop();
    }

    public int top() {
        return items.peek();
    }

    public int getMin() {
        return minimums.peek();
    }
}
